From 126671e73065e29c9582e47bc74774af2200fc8f Mon Sep 17 00:00:00 2001
From: Cameron Gutman <aicommander@gmail.com>
Date: Sun, 31 Aug 2025 16:05:15 -0500
Subject: [PATCH] avcodec/mfenc: add AVLowLatencyMode support

Set CODECAPI_AVLowLatencyMode when AV_CODEC_FLAG_LOW_DELAY is enabled on
the AVCodecContext. AVLowLatencyMode can acheive lower latency encoding
that may not be accessible using eAVScenarioInfo options alone.

Signed-off-by: Cameron Gutman <aicommander@gmail.com>
---
 libavcodec/mfenc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libavcodec/mfenc.c b/libavcodec/mfenc.c
index 6f79386404502..541f7fb9612af 100644
--- a/libavcodec/mfenc.c
+++ b/libavcodec/mfenc.c
@@ -876,6 +876,9 @@ static int mf_encv_output_adjust(AVCodecContext *avctx, IMFMediaType *type)

         if (c->opt_enc_scenario >= 0)
             ICodecAPI_SetValue(c->codec_api, &ff_CODECAPI_AVScenarioInfo, FF_VAL_VT_UI4(c->opt_enc_scenario));
+
+        if (avctx->flags & AV_CODEC_FLAG_LOW_DELAY)
+            ICodecAPI_SetValue(c->codec_api, &ff_CODECAPI_AVLowLatencyMode, FF_VAL_VT_UI4(1));
     }

     return 0;
